src/errors.ts

```diff
@@ -29,7 +29,12 @@ export class ReadOnlyError extends AzFuncTypeError {
 
 export function ensureErrorType(err: unknown): Error & Partial<AzFuncError> {
     if (err instanceof Error) {
-        return err;
+        const writable = Object.getOwnPropertyDescriptor(err, 'message')?.writable;
+        if (writable) {
+            return err;
+        } else {
+            return new Error(JSON.stringify(err));
```

Comment:
It's important that the value of `err.stack` is preserved on the new error as it's super helpful when debugging. The problem is that when you do `new Error()` the stack will change to this point in the code instead of the original point where err was thrown

In general, the value returned should be as close as possible to the original, because we don't know which properties might exist on the original object or if they're important

Reply:
If you're having troubles figuring out how to accomplish all that, I want to point out that we don't _have_ to make `message` settable. We could take a more reactive approach and somehow ignore errors if setting fails. The original error/stack is much more important that any additional text we're adding IMO

Reply:
I made some changes to my initial approach. Up until now I have avoided the reactive approach because I did not want to pass the responsibility to the caller but also because it would involve some ugly decisions like whether to put a try/catch inside another try/catch.

Also, before I forget to mention this, we also have `ensureErrorType` in the library repo: https://github.com/Azure/azure-functions-nodejs-library/blob/de87cbe25e05af90c2109c63e1e790dd252ebb18/src/errors.ts#L30. Whatever we decide to do here in the worker, we should probably copy over there too.

Reply:
Yes, please copy the change to the library once this is approved

Reply:
I will keep this thread open as a reminder. Once you approve the implementation here, I will open that new PR as well.

Reply:
Link to companion PR: https://github.com/Azure/azure-functions-nodejs-library/pull/220